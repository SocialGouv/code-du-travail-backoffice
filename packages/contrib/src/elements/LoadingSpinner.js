// https://loading.io/css/

import { keyframes } from "@emotion/core";
import styled from "@emotion/styled";
import React from "react";

const Container = styled.div`
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
`;

const ripple = keyframes`
  0% {
    height: 0;
    left: 28px;
    opacity: 1;
    top: 28px;
    width: 0;
  }
  100% {
    height: 58px;
    left: -1px;
    opacity: 0;
    top: -1px;
    width: 58px;
  }
`;
const Spinner = styled.div`
  animation: ${ripple} 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
  border-radius: 50%;
  border: 4px solid ${p => p.color};
  opacity: 1;
  position: absolute;
`;
const SpinnerNext = styled(Spinner)`
  animation-delay: -0.5s;
`;

export default ({ color = "white", ...props }) => {
  return (
    <Container {...props}>
      <Spinner color={color} />
      <SpinnerNext color={color} />
    </Container>
  );
};
