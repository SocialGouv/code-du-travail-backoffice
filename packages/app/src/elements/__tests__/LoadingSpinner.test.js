import React from "react";

import LoadingSpinner from "../LoadingSpinner";

describe("elements/<LoadingSpinner />", () => {
  it(`should pass`, () => {
    const $loadingSpinner = testRender(<LoadingSpinner size="8rem" />);
    const $spinner = $loadingSpinner.children[0];
    const $spinnerNext = $loadingSpinner.children[1];

    expect($loadingSpinner).toMatchSnapshot();

    expect($spinner).toHaveTestRenderedProp("color", "var(--color-text-blue)");
    expect($spinnerNext).toHaveTestRenderedProp("color", "var(--color-text-blue)");

    expect($spinner).toHaveStyleRule("border", "4px solid var(--color-text-blue)");
  });

  it(`should pass with {color} = "red"`, () => {
    const $loadingSpinner = testRender(<LoadingSpinner color="red" />);
    const $spinner = $loadingSpinner.children[0];
    const $spinnerNext = $loadingSpinner.children[1];

    expect($spinner).toHaveTestRenderedProp("color", "red");
    expect($spinnerNext).toHaveTestRenderedProp("color", "red");

    expect($spinner).toHaveStyleRule("border", "4px solid red");
  });
});
