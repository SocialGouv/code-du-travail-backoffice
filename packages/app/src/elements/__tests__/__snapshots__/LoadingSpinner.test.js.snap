// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<LoadingSpinner /> should pass 1`] = `
@keyframes animation-0 {
  0% {
    height: 0;
    left: 28px;
    opacity: 1;
    top: 28px;
    width: 0;
  }

  100% {
    height: 58px;
    left: -1px;
    opacity: 0;
    top: -1px;
    width: 58px;
  }
}

@keyframes animation-0 {
  0% {
    height: 0;
    left: 28px;
    opacity: 1;
    top: 28px;
    width: 0;
  }

  100% {
    height: 58px;
    left: -1px;
    opacity: 0;
    top: -1px;
    width: 58px;
  }
}

.emotion-4 {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}

.emotion-0 {
  -webkit-animation: animation-0 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: animation-0 1s cubic-bezier(0,0.2,0.8,1) infinite;
  border-radius: 50%;
  border: 4px solid white;
  opacity: 1;
  position: absolute;
}

.emotion-2 {
  -webkit-animation: animation-0 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: animation-0 1s cubic-bezier(0,0.2,0.8,1) infinite;
  border-radius: 50%;
  border: 4px solid white;
  opacity: 1;
  position: absolute;
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

<div
  className="emotion-4 emotion-5"
  size="8rem"
>
  <div
    className="emotion-0 emotion-1"
    color="white"
  />
  <div
    className="emotion-2 emotion-3"
    color="white"
  />
</div>
`;
